### PR TITLE
feat: error handler improvements

### PR DESCRIPTION
**Pull request by csviri**

### CODE REVIEW

File: operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/Controller.java [hunk 1]
```diff
@@ -60,32 +60,36 @@ public Controller(Reconciler<P> reconciler,
...
-            return reconciler.cleanup(resource, context);
-          }
-        });
+            @Override
+            public DeleteControl execute() {
+              return reconciler.cleanup(resource, context);
+            }
+          });
+    } catch (Exception e) {
+      throw new IllegalStateException(e);
```
Comment (metacosm) [inline operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/Controller.java:87]: Why `IllegalStateException`?

File: micrometer-support/src/main/java/io/javaoperatorsdk/operator/monitoring/micrometer/MicrometerMetrics.java [hunk 2]
```diff
@@ -33,7 +33,13 @@ public MicrometerMetrics(MeterRegistry registry) {
             .publishPercentileHistogram()
             .register(registry);
     try {
-      final var result = timer.record(execution::execute);
+      final var result = timer.record(() -> {
+        try {
+          return execution.execute();
+        } catch (Exception e) {
+          throw new IllegalStateException(e);
```
Comment (metacosm) [inline micrometer-support/src/main/java/io/javaoperatorsdk/operator/monitoring/micrometer/MicrometerMetrics.java:40]: Why throw an `IllegalStateException` here?

File: operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/Controller.java (hunk 1 above)
Comment (csviri) [reply operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/Controller.java:87]: Changed it to `OperatorException`.

File: micrometer-support/src/main/java/io/javaoperatorsdk/operator/monitoring/micrometer/MicrometerMetrics.java (hunk 2 above)
Comment (csviri) [reply micrometer-support/src/main/java/io/javaoperatorsdk/operator/monitoring/micrometer/MicrometerMetrics.java:40]: Changed it to `OperatorException`.

**Pull request merged**